moved context demo fails with ee10 environment

**Jetty version(s)**
12.1.x

**Jetty Environment**
ee10

**Description**
```
2025-08-06 13:16:19.349:WARN :oejd.DeploymentScanner:main: Failed to to perform action DEPLOY on oejd.DeploymentScanner$PathsApp@9d77da2b[demo-moved-context|ADDED, env=null, mainPath=/home/gregw/src/jetty-12.1.x/jetty-home/target/jetty-home/base/webapps/demo-moved-context.xml, paths=[/home/gregw/src/jetty-12.1.x/jetty-home/target/jetty-home/base/webapps/demo-moved-context.xml|ADDED], contextHandler=<unset>]
java.lang.IllegalArgumentException: Object of class 'org.eclipse.jetty.server.handler.MovedContextHandler' is not of type 'org.eclipse.jetty.ee10.webapp.WebAppContext'. Object Class and type Class are from different loaders.
        at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:481)
        at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:398)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyXml(StandardContextHandlerFactory.java:183)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyEnvironmentXml(StandardContextHandlerFactory.java:291)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.newContextHandler(StandardContextHandlerFactory.java:124)
        at org.eclipse.jetty.deploy.DeploymentScanner.performActions(DeploymentScanner.java:851)
        at org.eclipse.jetty.deploy.DeploymentScanner.pathsChanged(DeploymentScanner.java:589)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:918)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:817)
        at org.eclipse.jetty.util.Scanner.startScanning(Scanner.java:706)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:678)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.DeploymentScanner.doStart(DeploymentScanner.java:736)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.server.Server.start(Server.java:652)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:545)
        at org.eclipse.jetty.server.Server.doStart(Server.java:593)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:2108)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.eclipse.jetty.start.Main.invokeMain(Main.java:312)
        at org.eclipse.jetty.start.Main.start(Main.java:603)
        at org.eclipse.jetty.start.Main.main(Main.java:82)
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.eclipse.jetty.start.Main.invokeMain(Main.java:312)
        at org.eclipse.jetty.start.Main.start(Main.java:603)
        at org.eclipse.jetty.start.Main.main(Main.java:82)
Caused by: java.lang.IllegalArgumentException: Object of class 'org.eclipse.jetty.server.handler.MovedContextHandler' is not of type 'org.eclipse.jetty.ee10.webapp.WebAppContext'. Object Class and type Class are from different loaders.
        at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:481)
        at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:398)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyXml(StandardContextHandlerFactory.java:183)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyEnvironmentXml(StandardContextHandlerFactory.java:291)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.newContextHandler(StandardContextHandlerFactory.java:124)
        at org.eclipse.jetty.deploy.DeploymentScanner.performActions(DeploymentScanner.java:851)
        at org.eclipse.jetty.deploy.DeploymentScanner.pathsChanged(DeploymentScanner.java:589)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:918)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:817)
        at org.eclipse.jetty.util.Scanner.startScanning(Scanner.java:706)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:678)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.DeploymentScanner.doStart(DeploymentScanner.java:736)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.server.Server.start(Server.java:652)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:545)
        at org.eclipse.jetty.server.Server.doStart(Server.java:593)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:2108)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 4 more
java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:118)
        at java.base/java.lang.reflect.Method.invoke(Method.java:580)
        at org.eclipse.jetty.start.Main.invokeMain(Main.java:312)
        at org.eclipse.jetty.start.Main.start(Main.java:603)
        at org.eclipse.jetty.start.Main.main(Main.java:82)
Caused by: java.lang.IllegalArgumentException: Object of class 'org.eclipse.jetty.server.handler.MovedContextHandler' is not of type 'org.eclipse.jetty.ee10.webapp.WebAppContext'. Object Class and type Class are from different loaders.
        at org.eclipse.jetty.xml.XmlConfiguration$JettyXmlConfiguration.configure(XmlConfiguration.java:481)
        at org.eclipse.jetty.xml.XmlConfiguration.configure(XmlConfiguration.java:398)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyXml(StandardContextHandlerFactory.java:183)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.applyEnvironmentXml(StandardContextHandlerFactory.java:291)
        at org.eclipse.jetty.deploy.StandardContextHandlerFactory.newContextHandler(StandardContextHandlerFactory.java:124)
        at org.eclipse.jetty.deploy.DeploymentScanner.performActions(DeploymentScanner.java:851)
        at org.eclipse.jetty.deploy.DeploymentScanner.pathsChanged(DeploymentScanner.java:589)
        at org.eclipse.jetty.util.Scanner.reportDifferences(Scanner.java:918)
        at org.eclipse.jetty.util.Scanner.scan(Scanner.java:817)
        at org.eclipse.jetty.util.Scanner.startScanning(Scanner.java:706)
        at org.eclipse.jetty.util.Scanner.doStart(Scanner.java:678)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.deploy.DeploymentScanner.doStart(DeploymentScanner.java:736)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.start(ContainerLifeCycle.java:170)
        at org.eclipse.jetty.server.Server.start(Server.java:652)
        at org.eclipse.jetty.util.component.ContainerLifeCycle.doStart(ContainerLifeCycle.java:121)
        at org.eclipse.jetty.server.Handler$Abstract.doStart(Handler.java:545)
        at org.eclipse.jetty.server.Server.doStart(Server.java:593)
        at org.eclipse.jetty.util.component.AbstractLifeCycle.start(AbstractLifeCycle.java:93)
        at org.eclipse.jetty.xml.XmlConfiguration.main(XmlConfiguration.java:2108)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
        ... 4 more

```

**How to reproduce?**

```bash
java -jar ../start.jar --add-modules=demo-moved-context,ee10-environment
java -jar ../start.jar
```

--- a/jetty-home/src/main/resources/modules/demo-moved-context.mod
+++ b/jetty-home/src/main/resources/modules/demo-moved-context.mod
@@ -14,6 +14,7 @@ core-deploy
 
 [files]
 basehome:modules/demo.d/demo-moved-context.xml|webapps/demo-moved-context.xml
+basehome:modules/demo.d/demo-moved-context.properties|webapps/demo-moved-context.properties
 basehome:modules/demo.d/moved/index.html|webapps/moved/index.html
 basehome:modules/demo.d/moved/demo.css|webapps/moved/demo.css
 basehome:modules/demo.d/moved/small_powered_by.gif|webapps/moved/small_powered_by.gif

--- a/jetty-home/src/main/resources/modules/demo.d/demo-moved-context.properties
+++ b/jetty-home/src/main/resources/modules/demo.d/demo-moved-context.properties
@@ -1,0 +1,1 @@
+environment: core

